api/src/main/java/io/minio/MinioClient.java

```diff
@@ -3807,6 +3815,25 @@ private long getAvailableSize(Object data, long expectedReadSize)
     return totalBytesRead;
   }
 
+  private void fillData(ByteArrayOutputStream buffer, Object data, long size) throws IOException {
```

Comment:
Since this looks like a generic method to fill a given output stream, can it be moved to some utility class outside `MinioClient`?

Reply:
It is not generic method, it is specifically designed and used to `MinioClient`

Reply:
We can still have something like `MinioClientUtil`. The `MinioClient.java` is too huge for comfort (5000+ lines)

Reply:
This is different issue, needs to be addressed in different PR.

Reply:
Your call. IMHO it's a good idea to start straight away, especially for new methods that are being introduced.